sklearn/datasets/openml.py

```diff
@@ -263,6 +266,103 @@ def _convert_arff_data(arff_data, col_slice_x, col_slice_y, shape=None):
         raise ValueError('Unexpected Data Type obtained from arff.')
 
 
+def _feature_to_dtype(feature):
+    """Map feature to dtype for pandas DataFrame
+    """
+    if feature['data_type'] == 'string':
+        return object
+    elif feature['data_type'] == 'nominal':
+        return 'category'
+    # only numeric, integer, real are left
+    elif (feature['number_of_missing_values'] != '0' or
+          feature['data_type'] in ['numeric', 'real']):
+        # cast to floats when there are any missing values
+        return np.float64
+    elif feature['data_type'] == 'integer':
+        return np.int64
+    raise ValueError('Unsupported feature: {}'.format(feature))
+
+
+def _chunk_iterable(seq, chunksize):
```

Comment:
Any reason this (from [SO](https://stackoverflow.com/questions/312443/how-do-you-split-a-list-into-evenly-sized-chunks))

```py
def chunks(l, n):
    """Yield successive n-sized chunks from l."""
    for i in range(0, len(l), n):
        yield l[i:i + n]
```

cannot be used?

Reply:
cannot slice a generator

Reply:
`islice`?

Reply:
Something like

```py
def chunks(l, n):    
    from itertools import islice    
    start, end = 0, n    
    while True:    
        out = tuple(islice(l, start, end))    
        if out:    
            yield out    
            start += n    
            end += n    
        else:    
            return 
```

I'm pretty sure there are more pythonic versions though